### PR TITLE
add support for new Token Metadata delegate

### PR DESCRIPTION
**Pull request by samuelvanderwaal**
Adds support for the new Token Metadata delegate type.

### CODE REVIEW

File: programs/bubblegum/program/tests/collection.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#![cfg(feature = "test-bpf")]
+pub mod utils;
```
Comment (danenbm) [inline programs/bubblegum/program/tests/collection.rs:2]: Awesome thank you for adding these tests

File: programs/bubblegum/program/Cargo.toml [hunk 2]
```diff
@@ -35,6 +35,4 @@ spl-concurrent-merkle-tree = "0.1.2"
 spl-merkle-tree-reference = "0.1.0"
 spl-noop = { version = "0.1.3", features = ["no-entrypoint"] }
 mpl-token-auth-rules = { version = "1.1", features = ["no-entrypoint"] }
-
-[profile.release]
-overflow-checks = true
```
[the remark this answers is not in the document]
Comment (samuelvanderwaal) [reply programs/bubblegum/program/Cargo.toml:40]: They're still on [here](https://github.com/metaplex-foundation/mpl-bubblegum/blob/1141918f03512a2d743266b5e888512ccf7e4686/programs/bubblegum/Cargo.toml#L1). Only profile settings at the root level are followed so having it in this sub-dir's toml file had no effect.

[the remark this answers is not in the document]
Comment (samuelvanderwaal) [reply programs/bubblegum/program/Cargo.toml:40]: Profile settings have to be at the workspace level if a workspace exists.

**Review by danenbm — APPROVED** [review]
LGTM

**Pull request merged**